src/networking/TcpHelperClasses.cpp

```diff
@@ -29,10 +29,7 @@ void TcpRxConnection::start ()
     connection_state_t exp = connection_state_t::halted;
     if (state.compare_exchange_strong (exp, connection_state_t::receiving))
     {
-        if (receiving == false)
```

Comment:
the probably should be left as it was.  The reason for this is because receiving is an atomic and reads are alot less expensive then writes so we don't want to do a write unless we really need to.